sys/arch/detect.S

```diff
@@ -390,10 +390,14 @@ SYM(detect_fpu):
 	dc.l	0x4e7a0808	// movec pcr,d0
 	swap	d0
 	cmp.w	#0x0431,d0	// "broken" 68LC/EC060
-	beq.s	no60
-	swap	d0
+	bne.b	rc60
+	clr.l   d0
+	bra.s   fexit
+
+rc60:	swap	d0
 	bclr	#0x01,d0
 	dc.l	0x4e7b0808	// movec d0,pcr
+	clr.l   d0
```

Comment:
What is this instruction good for? `d0` is later overwritten by `0x10.l`, isn't it?

Reply:
After a second look... ok, in case the 060 FPU is present but not working (throwing an exception) this will ensure the proper return value.

Reply:
> After a second look... ok, in case the 060 FPU is present but not working (throwing an exception) this will ensure the proper return value.

If it is throwing an exception, it won't get to the clr.l (unless someone catches that exception and emulates the movec)

Reply:
I wouldn't expect it to throw on movec but on the fnop below.